scene/2d/tile_map.cpp

```diff
@@ -3907,10 +3907,10 @@ void TileMap::draw_cells_outline(Control *p_control, const RBSet<Vector2i> &p_ce
 		} else {
 			if (tile_set->get_tile_offset_axis() == TileSet::TILE_OFFSET_AXIS_HORIZONTAL) {
 				if (shape == TileSet::TILE_SHAPE_ISOMETRIC) {
-					DRAW_SIDE_IF_NEEDED(TileSet::CELL_NEIGHBOR_BOTTOM_RIGHT_SIDE, 3, 4);
-					DRAW_SIDE_IF_NEEDED(TileSet::CELL_NEIGHBOR_BOTTOM_LEFT_SIDE, 2, 3);
+					DRAW_SIDE_IF_NEEDED(TileSet::CELL_NEIGHBOR_BOTTOM_RIGHT_SIDE, 2, 3);
```

Comment:
This seems wrong. If the shape is the same for both mode, then you should not have different code path here. You should move that our of the if (like `TILE_SHAPE_SQUARE`) and make sure each side corresponds to the correct polygon index.

You can test that by selecting tiles in the TileMap if I am not mistaking, if the code is wrong selecting tiles on the TileMap will not work correctly.

Reply:
I tested before that it kept drawing the outlines correctly after my changes and it was working fine. But you're right, there's no longer need to swap xy for isometric tile polygon in case of different offset axis. And not doing so will make this branching not needed indeed.